Tcp midstream fin 5437 v1

catenacyber opened the pull request:
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5437

Describe changes:
- relax json schema for alert metadata
- pick up midstream session even with TCP FIN flag set

suricata-verify-pr: 1051

suricata-qa commented:
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 128237 | 104015 | 81.11% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 53235 | 0.01 | 

Pipeline 11300

victorjulien commented:
Wonder if:
1. this should be optional
2. perhaps only happen for FIN + data packets

catenacyber commented:
> Wonder if:
> 
> 1. this should be optional

I have no idea. Tell me what you prefer

> 2. perhaps only happen for FIN + data packets

Sounds relevant, added a check.

Replaced by https://github.com/OISF/suricata/pull/8445
